### PR TITLE
Add range-v3

### PR DESCRIPTION
**Pull request by bluescarni**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by bluescarni** [comment]
@conda-forge/help-c-cpp this should now be ready for review.

range-v3 is a header-only C++ library implementing the C++20 ranges proposal for earlier versions of the language (C++14 and C++17). It has no dependencies apart from the standard library.

See also #7824 .

**Comment by bluescarni** [comment]
@conda-forge/help-c-cpp could I get a review please?

File: recipes/range-v3/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+source:
+  url: https://github.com/ericniebler/range-v3/archive/refs/tags/{{ version }}.tar.gz
+  sha256: 015adb2300a98edfceaf0725beec3337f542af4915cec4d0b89fa0886f4ba9cb
+
+build:
+  number: 0
+  run_exports:
+    # Let's keep this conservative.
+    - {{ pin_subpackage('range-v3', max_pin='x.x') }}
```
Comment (apmorton) [inline recipes/range-v3/meta.yaml:15]: range-v3 is a header only library - applications built against it should not need it installed at runtime.

File: recipes/range-v3/bld.bat [hunk 2]
```diff
@@ -0,0 +1,21 @@
+mkdir build
+cd build
+
+cmake ^
+    -G "Visual Studio 16 2019" -A x64 ^
```
Comment (isuruf) [inline recipes/range-v3/bld.bat:5]:
```suggestion
```

**Comment by bluescarni** [comment]
@apmorton @isuruf thanks for the review, I have addressed the comments.

**Comment by bluescarni** [comment]
@isuruf @apmorton do you think this is ready to be merged?

**Comment by apmorton** [comment]
LGTM, but I'm not a maintainer here - just another user interested in having range-v3 on conda-forge.

**Comment by bluescarni** [comment]
@conda-forge/help-c-cpp could this be merged please?

File: recipes/range-v3/bld.bat [hunk 3]
```diff
@@ -0,0 +1,20 @@
+mkdir build
+cd build
+
+cmake -G "Visual Studio 16 2019" -A x64 ^
```
Comment (isuruf) [inline recipes/range-v3/bld.bat:4]:
```suggestion
cmake ^
```

Comment (bluescarni) [reply recipes/range-v3/bld.bat:4]: Ahh so you meant to complete remove the ``-G`` option?

File: recipes/range-v3/bld.bat [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    -DRANGES_BUILD_CALENDAR_EXAMPLE=NO ^
+    -DRANGES_DEBUG_INFO=NO ^
+    -DRANGE_V3_DOCS=NO ^
+    -DRANGE_V3_EXAMPLES=NO ^
+    -DRANGE_V3_TESTS=YES ^
+    ..
+
+cmake --build . --config Release
+
+ctest --output-on-failure -j${CPU_COUNT} -V -C Release
```
Comment (isuruf) [inline recipes/range-v3/bld.bat:18]:
```suggestion
ctest --output-on-failure -j%CPU_COUNT% -V -C Release
```

**Review by isuruf — APPROVED** [review]

**Comment by bluescarni** [comment]
@isuruf any idea why the build with NMake is now failing in Windows?

**Comment by isuruf** [comment]
See https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/4357

**Pull request merged**